### PR TITLE
Refactor search_documents Method Into Search Base.rb Class

### PR DESCRIPTION
**Pull request by mstruve**
## What type of PR is this? (check all applicable)
- [x] Refactor

## Description
- Move search_documents method into `Search::Base` class
- Update all query builders to use keyword arguments for params
- Remove user_id argument from ChatChannelMembership query builder and add to params
- Choosing to ignore the leftover duplicate code for now. You can't have it all CodeClimate, that's not how this works. 

## Related Tickets & Documents
https://github.com/thepracticaldev/dev.to/projects/6#card-36211320

## ~Added~ Updated tests?
- [x] yes

alt_text

### CODE REVIEW

File: app/services/search/base.rb [hunk 1]
```diff
@@ -41,8 +41,21 @@ def document_count
         Search::Client.count(index: self::INDEX_ALIAS).dig("count")
       end
 
+      def search_documents(params:)
+        set_query_size(params)
+        query_hash = "Search::QueryBuilders::#{name.gsub('Search::', '')}".safe_constantize.new(params: params).as_hash
```
Comment (atsmith813) [inline app/services/search/base.rb:46]: I think `demodulize` might clean this up a little bit:

```ruby
query_hash = "Search::QueryBuilders::#{name.demodulize}".safe_constantize.new(params: params).as_hash
```

A little [trick](https://github.com/thepracticaldev/dev.to/pull/6801#discussion_r397065488) picked up from @rhymes via @citizen428 😄

Comment (mstruve) [reply app/services/search/base.rb:46]: Oooooh, thats 🔥

File: app/services/search/base.rb [hunk 2]
```diff
@@ -41,8 +41,21 @@ def document_count
         Search::Client.count(index: self::INDEX_ALIAS).dig("count")
       end
 
+      def search_documents(params:)
```
Comment (rhymes) [inline app/services/search/base.rb:44]: could we add a default of `{}` here? or you don't want to risk that by chance someone is able to search all documents?

Comment (mstruve) [reply app/services/search/base.rb:44]: I think it's better to require params to prevent that from accidentally happening.

**Pull request merged**